.travis.yml

```diff
@@ -1,15 +1,125 @@
+#
+# See https://github.com/ethereum/webthree-umbrella/blob/develop/.travis.yml
+#
+# ------------------------------------------------------------------------------
+# This file is part of cpp-ethereum.
+#
+# cpp-ethereum is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cpp-ethereum is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>
+#
+# (c) 2016 cpp-ethereum contributors.
+#------------------------------------------------------------------------------
+
 language: cpp
-cache: ccache
+branches:
+    only:
+        - develop
+        - release
+        - standalone
+    except:
+        - /develop-v[0-9]/
 matrix:
     include:
         - os: linux
           dist: trusty
           sudo: required
-          env: TRAVIS_BUILD_TYPE=Debug
+          compiler: gcc
+          services:
+              - docker
+          before_install:
+              - docker pull trzeci/emscripten:sdk-tag-1.35.4-64bit
+          env:
+              - TRAVIS_BUILD_TYPE=RelWithDebInfo
+              - ZIP_SUFFIX=ubuntu-trusty
+        - os: osx
+          osx_image: beta-xcode6.2
+          env:
+              - TRAVIS_BUILD_TYPE=RelWithDebInfo
+              - ZIP_SUFFIX=osx-mavericks
+        - os: osx
+          osx_image: xcode7.1
+          env:
+              - TRAVIS_BUILD_TYPE=RelWithDebInfo
+              - ZIP_SUFFIX=osx-yosemite
         - os: osx
           osx_image: xcode7.3
-          env: TRAVIS_BUILD_TYPE=Debug
+          env:
+              # The use of Debug config here ONLY for El Capitan is a workaround for "The Heisenbug"
+              # See https://github.com/ethereum/webthree-umbrella/issues/565
+              - TRAVIS_BUILD_TYPE=Debug
+              - ZIP_SUFFIX=osx-elcapitan
 git:
     depth: 2
-before_install: ./install_dependencies.sh
-script: ./fetch_umbrella_build_and_test.sh
\ No newline at end of file
+install:
+    - ./scripts/install_deps.sh
+before_script:
+    - ./scripts/build_emscripten.sh
+    - mkdir -p build
+      && cd build
+      && cmake .. -DCMAKE_BUILD_TYPE=$TRAVIS_BUILD_TYPE
+      && make -j2
+      && cd ..
+      && ./scripts/release.sh $ZIP_SUFFIX
+script:
+    # There are a variety of reliability issues with the Solidity unit-tests at the time of
+    # writing, so we're actually running them 5 times in a row, to try to flush all of these
+    # issues out as quickly as possible.  See https://github.com/ethereum/solidity/issues/769    
+    - cd $TRAVIS_BUILD_DIR && ./scripts/tests.sh
```

Comment:
I don't think this is what we want. The instabilities are instabilities of eth, so the test run should fail if and only if all the runs fail. Currently, it fails if and only if a single run fails.